lib/elixir/lib/kernel.ex

```diff
@@ -3543,17 +3543,28 @@ defmodule Kernel do
       [] when not in_module? ->
         false
 
-      [head | tail] = list when not in_module? ->
-        in_var(in_module?, left, &in_list(&1, head, tail, expand, list, in_module?))
+      [] ->
+        quote(do: :lists.member(unquote(left), []))
```

Comment:
Can we just merge this clause with the previous one? and do

```elixir
      [] ->
￼	        false
```

Reply:
Yes, we could.

Reply:
Actually, we cannot. It would raise unused variable warnings. And not execute the left side at all. We can probably compile it to this though:

    quote do
      _ = unquote(left)
    end

Then it should safely apply. But yeah, they can't be merged, sorry.